ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -598,12 +598,18 @@ let on_request :
   | TextDocumentColor _ -> now []
   | TextDocumentColorPresentation _ -> now []
   | TextDocumentHover req ->
-    later
-      (fun (_ : State.t) () -> Hover_req.handle rpc req Hover_req.Default)
-      ()
+    let mode =
+      match state.configuration.data.extended_hover with
+      | Some { enable = Some true } -> Hover_req.Extended_variable
+      | Some _ | None -> Hover_req.Default
+    in
+    later (fun (_ : State.t) () -> Hover_req.handle rpc req mode) ()
   | TextDocumentReferences req -> later references req
   | TextDocumentCodeLensResolve codeLens -> now codeLens
-  | TextDocumentCodeLens req -> later text_document_lens req
+  | TextDocumentCodeLens req -> (
+    match state.configuration.data.codelens with
+    | Some { enable = Some true } -> later text_document_lens req
+    | None | Some _ -> now [])
```

Comment:
Does None here mean, that if we update extended_hover without codelens, they will get disabled?
This logic is a bit tricky, but it will be resolved if we use boolean defaults directly.

Reply:
My mistake. `None` should be in the first case match case.

By default `codelens` is enabled.

Reply:
Done 175d5e64a0df1d98ad4b31a3a7b5509973d8b554